### ISSUE TITLE
Unticking "Use TLB" wipes save file data for some games upon booting them.

### ISSUE THREAD

**Issue by retrobenny** [project64/project64#1359]
Confirmed that it keep happening on Banjo-Tooie but it may also occur on other games like DK64 as well.

This is not a problem on older PJ versions like 1.7 and unticking this option is needed for some of the extra fun codes to work without errors/crashing.

It even wipes the save file data without any cheats enabled,so its not my codes causing it.
I have even tried re-copying a confirmed working save and it still gets wiped again.

**Comment by cxd4** [comment]
```c
if (options->TLB == true && supporter == "retroben") {
    wipe_save_file();
    mess_with_cheat_offsets();
    fuck_with_retroben();
}
```

**Comment by retrobenny** [comment]
Hah!

TLB=Torture Lil' Ben

**Comment by retrobenny** [comment]
I have an idea for a solution!

Maybe an option can be made for a hacked version of TLB that disables the errors/exceptions/crashes/freezes/etc. triggered by TLB when on Recompiler,that would allow more codes to work without completely disabling TLB.

The name for it could be "Ignore TLB Exceptions".

**Comment by LegendOfDragoon** [comment]
You shouldn't disable TLB for games that use it. A lot of Rare games use em.

**Comment by retrobenny** [comment]
Seemed like both Tooie and DK64 ran fine with TLB off since an older PJ that doesn't attack save files,but I wonder what both games use TLB for. Having that middle solution would be very nice.

**Comment by retrobenny** [comment]
Got some videos of what disabling TLB allows with the non-cheat glitch.
The player doesn't work in Chrome for me and its YT player so it may harm video quality/clarity,I wished they would bring back the "Google Video" player instead since its quality,performance,and fps was crystal clear from a video of Sonic CD being played back when the video site still existed.

1. With TLB enabled.

https://drive.google.com/open?id=0B1nr_75cEfQOYTd5dFphU2h0SEU

2. With TLB disabled.

https://drive.google.com/open?id=0B1nr_75cEfQOaDJBTFE0TEtQbmc

Clear examples of a freeze without any Gameshark codes when TLB is on but continuation with TLB disabled due to it avoiding the extra exceptions that halt the game.

**Comment by retrobenny** [comment]
Turns out its not TLB toggling causing the save to be wiped.
Saving in the game on that same version had it staying intact on the next time its booted when re-opening PJ.

Its actually a compatibility issue with saves made on other versions of PJ and other emulators.
These saves are failing to load in newer PJ builds.
I managed to get my save file data to write by using a savestate from an older PJ inside of the new PJ build and using save and quit,then it started staying intact!
This part was done with TLB disabled the entire time! Closing issue to make a proper one.
